feat(ui): Add notification system for rss feeds

sumit-tft opened the pull request:
### Context

This PR introduces a notification system to the Prowler UI 
It adds a bell icon in the top-right corner, fetching alerts from an external RSS feed. 
A red dot appears on the bell when new notifications arrive.

### Description

- Created `feeds-server.tsx` (Server Component) for initial data fetch.
- Created `feeds-client.tsx` (Client Component) to handle UI updates.
- Created `feeds.ts` (Action) to fetch the feeds
- The notification system loads **only on cloud domains** (e.g., `cloud.prowler.com`).
- Added styling for the notification bell and dropdown in the `ui` or `dashboard` directory.
- Ensured that UI updates are smooth and efficient.
- Used `rss-parser` library to parse rss feeds

### Steps

1. Checkout `feature/add-notification-system` branch.
2. Run the docker containers
3. Make sure the new npm package `rss-parser` installed
4. Open `http://localhost:3000` in your browser
5. Verify that the notification bell is in the top-right corner
6. Check if notifications appear when fetching from the RSS feed(currently showing with mock data)

Note: To use it in development mode please ensure that `NEXT_PUBLIC_IS_CLOUD_ENV` should be set as true

sumit-tft committed:
feat(ui): Add notification system for rss feeds

sumit-tft committed:
srn Updated the logic to show the notification icon

sumit-tft committed:
srn Fixed lint issue

paabloLC commented on ui/components/ui/nav-bar/navbar.tsx:
```diff
@@ -26,6 +27,7 @@ export function Navbar({ title, icon, user }: NavbarProps) {
           <h1 className="text-sm font-bold text-default-700">{title}</h1>
         </div>
         <div className="flex flex-1 items-center justify-end gap-3">
+          {process.env.NEXT_PUBLIC_IS_CLOUD_ENV === "true" && <FeedsServer />}
```
Hi @sumit-tft , we want this to be visible in Prowler as well. Could you please remove this condition?

paabloLC reviewed (changes requested):
I left some comments, @sumit-tft. Please take a look when you have a moment. Thank you

sumit-tft committed:
Merge branch 'upstream/master' into feature/add-notification-system

# Conflicts:
#	ui/components/icons/Icons.tsx

sumit-tft commented on ui/actions/feeds/feeds.tsx:
```diff
@@ -0,0 +1,25 @@
+import Parser from "rss-parser";
+
+// TODO: Need to use the actual feed url
+const RSS_FEED_URL = "https://prowler.com/blog/rss";
```
Updated

sumit-tft commented on ui/components/feeds/feeds-client.tsx:
```diff
@@ -0,0 +1,77 @@
+            variant="outline"
+            className={cn("relative", "rounded-full", "bg-transparent", "p-2")}
+          >
+            <Icon size={18} />
```
![image](https://github.com/user-attachments/assets/dc8aaea1-003b-4b7c-8d07-c6e5ad817574)

Updated, please let me know if any changes are required :)

sumit-tft committed:
srn Lint fixes

sumit-tft committed:
Merge branch 'upstream/master' into feature/add-notification-system

jfagoagas commented:
Hey @sumit-tft, I was reviewing the functionality of the RSS feed and I noticed the following:
- We should not show the bell's red dot if no RSS feed configured or if there is nothing new
![Screenshot 2025-05-13 at 08 25 53](https://github.com/user-attachments/assets/eacb7e5f-f887-42c8-a407-5e0f561c4cce)
- If `RSS_FEED_URL` is empty or not set the UI logs are showing the following. Could you please review it? I think we should not do anything in that case.
```
Error fetching RSS feed: AggregateError [ECONNREFUSED]: 
    at internalConnectMultiple (node:net:1139:18)
    at afterConnectMultiple (node:net:1712:7)
    at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
  code: 'ECONNREFUSED',
  [errors]: [
    Error: connect ECONNREFUSED ::1:80
        at createConnectionError (node:net:1675:14)
        at afterConnectMultiple (node:net:1705:16)
        at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
      errno: -61,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '::1',
      port: 80
    },
    Error: connect ECONNREFUSED 127.0.0.1:80
        at createConnectionError (node:net:1675:14)
        at afterConnectMultiple (node:net:1705:16)
        at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
      errno: -61,
      code: 'ECONNREFUSED',
      syscall: 'connect',
      address: '127.0.0.1',
      port: 80
    }
  ]
}
```
- When the RSS feed has something new I think we should remove `Comments on: Blog` and just show the blog post title, what do you think?
![Screenshot 2025-05-13 at 08 28 05](https://github.com/user-attachments/assets/e760cd24-7154-4b44-a4e2-5155509f07ed)

sumit-tft committed:
Merge branch 'upstream/master' into feature/add-notification-system

# Conflicts:
#	ui/package.json

sumit-tft committed:
Merge branch 'upstream/master' into feature/add-notification-system

sumit-tft committed:
srn PR comments

sumit-tft committed:
srn Added missing nullable

sumit-tft commented:
Hello, Thank you for reviewing this PR, I discussed with Pablo and made the changes 
 - Updated the color of the dot and also added a condition if we don't have any feeds than the dot will not appear(this is a temporary solution as of, we have to show it dynamically when data changes)
 - Thank you for catching this error :) I added a condition to avoid calling of an endpoint if not set
 - For the third point, I need some more data in the feed URL to determine how we get the result, currently we are getting one channel only, if data changes for that channel than it will update on feed also

sumit-tft commented:
I’ve resolved the conflicts and recreated the PR from the master branch ([PR #8394](https://github.com/prowler-cloud/prowler/pull/8394)). Closing this PR now
